internal/services/compute/managed_disk_resource.go

```diff
@@ -673,11 +694,13 @@ func resourceManagedDiskRead(d *pluginsdk.ResourceData, meta interface{}) error
 				d.Set("logical_sector_size", creationData.LogicalSectorSize)
 			}
 
-			imageReferenceID := ""
-			if creationData.ImageReference != nil && creationData.ImageReference.ID != nil {
-				imageReferenceID = *creationData.ImageReference.ID
+			if galleryImageReference := creationData.GalleryImageReference; galleryImageReference != nil {
+				setImageReference(d, *galleryImageReference)
+			} else if imageReference := creationData.ImageReference; imageReference != nil {
+				setImageReference(d, *imageReference)
+			} else {
+				setImageReference(d, compute.ImageDiskReference{})
```

Comment:
there's no need to split this method out, with a separate field for GalleryImageID this becomes simpler and can be inlined, which also moves this left:

```suggestion
			galleryImageId := ""
			imageId := ""
			if galleryImageReference := creationData.GalleryImageReference; galleryImageReference != nil && galleryImageReference.ID != nil {
				galleryImageId = *galleryImageReference.ID
			}
			if imageReference := creationData.ImageReference; imageReference != nil && imageReference.ID != nil {
				imageId = *imageReference.ID
			}
```

Reply:
Yeah since imgeReference.Lun is gone, have move the code out of the new method